### DISCUSSION TITLE
Question about .env files and EAS builds

### DISCUSSION THREAD

**Discussion by Spharian** [goatandsheep/react-native-dotenv#388] [Q&A]
Hello :)!

The push on the app store approaching for my app, I'm trying to understand how this package works with EAS build. Actually, I have these files on my projects:

`.env.development` for the keys used in dev mode, locally on my computer
`.env.production` for the keys used when doing `expo publish --clear`, for testing the app on real devices

I'm a bit confused. This `env.production` file won't be used in EAS build right? It's gonna switch to EAS secrets defined on my expo website under that link https://expo.dev/accounts/ACCOUNT_NAME/projects/PROJECT_NAME/secrets, or am I wrong?

Kind regards

**Comment by goatandsheep** [comment]
Make sure to keep your .env files in your .gitignore. It is a major security risk to commit. The EAS documentation says:

> To create app-specific secrets, navigate to the "Secrets" tab in your project dashboard.

**Comment by Spharian** [comment]
Thanks @goatandsheep but if I don't commit my `.env` files, how can I configure the env keys for the app on expo, for example? (not a standalone app but just expo for testing on device), as EAS is only used for standalone app builds.

**Reply by goatandsheep** [reply]
This library reads the variables from the environment, i.e. shell variables. EAS has a place to enter them and the library will read it

**Reply by Spharian** [reply]
Thanks for your fast answers :) When we use `expo publish` to run the app on Expo Go, it's not using EAS secrets variables. My question is: how to "send" these env variables with every publish without committing the file? I thought the point of `.env` files was to write keys/values to a single place, and let the package do the rest. I'd like to know how a basic project is organized for:

- Run the project locally, with specific keys (eg: `API_URL=http://localhost:8000`)
- Run the project on a staging server (Expo Go), without committing the .env files, with specific keys (eg: `API_URL=http://staging.app.com`)
- Run the project on a production server (standalone app), with specific keys (eg: `API_URL=http://production.app.com`) -> for this one I guess the EAS secrets can be used

**Comment by Spharian** [comment]
Ok so I managed to understand that the `.env` files do not need to be pushed and are used only during `expo publish --clear`.

My next question:
the `.env.production` file seems to be used in priority in Expo Go + TestFlight (maybe because of automatic update of the TestFlight installed app after `expo publish --clear`?). Is it 100% sure that the App-specific Secrets from expo.dev gonna be used in priority when I'll be building the final app for the App Store? Or should I use another file naming convention for my env files?

**Reply by goatandsheep** [reply]
You'll have to setup your environment in your app configuration on XCode https://github.com/goatandsheep/react-native-dotenv/wiki/Multi-env-troubleshooting#xcode
if that doesn't work, then there must be a caching issue of some sort. a lot of caching issues are to be fixed after Expo updates their Babel version https://github.com/expo/expo-cli/pull/4593
